### PR TITLE
094_孙政_第三课作业

### PR DESCRIPTION
**Pull request by Hadesonly**

### CODE REVIEW

**Comment by Hadesonly** [comment]
add C3 homework @h1994st @minzhi3

File: projects/lesson-3/C3_homework.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+contract K1 is A, B
+contract K2 is A, C
+contract Z is K1, K2
+```
+
+O = [O]
+mro(A) = [A,O]
+mro(B) = [B,O]
+mro(C) = [C,O]
+mro(K1) = [K1] + merge(mro(A), mro(B), [A,B]) = [K1] + merge([A,O], [B,O], [A,B]) = [K1,A] + merge([O], [B,O], [B]) + [K1,A,B] + merge([O],[O]) = [K1,A,B,O]
```
Comment (minzhi3) [inline projects/lesson-3/C3_homework.txt]: 在solidity里，多继承contract K1 is A, B {...} 这个意思是：合约K1
继承自合约B，合约A，其中B是最先被继承的合约("most-derived"), 继承链应该是K1 <--- B <--- A 见文档：
* https://solidity.readthedocs.io/en/develop/contracts.html#inheritance 

所以这里展开应该是
[K1] + merge(L[B],L[A],[B,A])

File: projects/lesson-3/C3_homework.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+mro[K2] = [K2] + merge(mro(A),mro(C),[A,C]) = ... = [K2,A,C,O]
```
Comment (minzhi3) [inline projects/lesson-3/C3_homework.txt]: 同理这里展开应该是
[K2] + merge(L[C],L[A],[C,A])

**Review by minzhi3 — COMMENTED** [review]
附加题有点问题，注意继承顺序

正确答案如下：
L(Z) = [Z] + merge(L(K2), L(K1), [K2, K1])
= [Z] + merge([K2, C, A, O], [K1, B, A, O], [K2, K1])
= [Z, K2] + merge([C, A, O], [K1, B, A, O], [K1])
= [Z, K2, C] + merge([A, O], [K1, B, A, O], [K1])
= [Z, K2, C, K1] + merge([A, O], [B, A, O])
= [Z, K2, C, K1, B] + merge([A, O], [A, O])
= [Z, K2, C, K1, B, A, O]

File: projects/lesson-3/C3_homework.txt (hunk 2 above)
Comment (Hadesonly) [reply projects/lesson-3/C3_homework.txt]: got it

**Comment by Hadesonly** [comment]
already fix

**Pull request closed without merging**